llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -756,24 +756,56 @@ processInputModule(std::unique_ptr<Module> M) {
       module_split::getSplitterByMode(module_split::ModuleDesc{std::move(M)},
                                       SplitMode, IROutputOnly,
                                       EmitOnlyKernelsAsEntryPoints);
-  const bool SplitByScope = ScopedSplitter->totalSplits() > 1;
-  Modified |= SplitByScope;
 
+  SmallVector<module_split::ModuleDesc, 8> TopLevelModules;
+  bool SplitByOptionalFeatures = false;
+
+  // FIXME: this check should be performed on all split levels
   if (DeviceGlobals)
     ScopedSplitter->verifyNoCrossModuleDeviceGlobalUsage();
 
+  while (ScopedSplitter->hasMoreSplits()) {
+    module_split::ModuleDesc MD = ScopedSplitter->nextSplit();
+
+    if (IROutputOnly || SplitMode == module_split::SPLIT_NONE) {
+      // We can't perform any kind of split
```

Comment:
```suggestion
      // We can't perform any kind of split.
```

Reply:
Applied in 0cc45b2c1a04551249551bf104bd99b559a55d83